Sources/Sentry/SentryClient.m

```diff
@@ -168,13 +169,37 @@ - (SentryId *)captureError:(NSError *)error
 - (SentryEvent *)buildErrorEvent:(NSError *)error
 {
     SentryEvent *event = [[SentryEvent alloc] initWithError:error];
-    NSString *errorCodeAsString = [NSString stringWithFormat:@"%ld", (long)error.code];
-    SentryException *sentryException = [[SentryException alloc] initWithValue:errorCodeAsString
-                                                                         type:error.domain];
-    event.exceptions = @[ sentryException ];
-    event.fingerprint = @[ error.domain, [NSString stringWithFormat:@"%ld", (long)error.code] ];
 
-    [self setUserInfo:error.userInfo withEvent:event];
+    NSMutableString *exceptionValue =
+        [NSMutableString stringWithFormat:@"Code: %ld", (long)error.code];
+
+    // We only want to add the NSLocalizedDescription if the error contains one. As
+    // NSError.localizedDescription returns a default string contructed from the domain and code if
+    // the user info dictionary doesn’t contain a value for NSLocalizedDescriptionKey, we don't use
+    // NSError.localizedDescription as it would just again append domain and code, which we already
+    // have in the SentryException.
+    NSString *localizedDescription = error.userInfo[NSLocalizedDescriptionKey];
```

Comment:
This is what we used to do and after customer feedback we decided to drop it.
Probably better to put this in `exception.data` or `exception.mechanism.data` only.